stl/inc/xmemory

```diff
@@ -1551,6 +1551,18 @@ namespace ranges {
         && _No_throw_forward_iterator<iterator_t<_Rng>>;
     // clang-format on
 
+    template <class _InIt, class _OutIt>
+    in_out_result<_InIt, _OutIt> _Copy_memcpy_common(
+        _InIt _IFirst, _InIt _ILast, _OutIt _OFirst, _OutIt _OLast) noexcept {

```

Comment:
The `reinterpret_cast`s herein can only be valid if `_InIt` and `_OutIt` are pointers. I think the signature should reflect that. Maybe have the function parameter types be `_InTy*` and `_OutTy*`?

Reply:
I have no objection to such a signature change, although I do note that there is existing code that follows the current pattern:
https://github.com/microsoft/STL/blob/47cd70315d560e4fe1ac699c0eb8ec8bfc5cb658/stl/inc/xutility#L4784-L4796

Reply:
Unsurprisingly, I'm not fond of that either. @miscco, feel free to file an issue to address both of these occurrences and commit this as-is.